src/dialogs/vkbasalt.py

```diff
@@ -0,0 +1,340 @@
+                vkbasalt_settings.smaa_edge_detection = self.smaa_edge_detection
+                vkbasalt_settings.smaa_corner_rounding = Gtk.Adjustment.get_value(self.spin_smaa_corner_rounding)
+                vkbasalt_settings.smaa_max_search_steps = Gtk.Adjustment.get_value(self.spin_smaa_max_search_steps)
+                vkbasalt_settings.smaa_max_search_steps_diagonal = Gtk.Adjustment.get_value(self.spin_smaa_max_search_steps_diagonal)
+
+            vkbasalt_settings.effects = tuple(effects)
+
+        else:
+            vkbasalt_settings.effects = False
+
+        if self.btn_lut_file_path:
+            vkbasalt_settings.lut_file_path = self.btn_lut_file_path
+
+        vkbasalt_settings.output = conf
+
+        parse(vkbasalt_settings)
+        self.close()
+        return GLib.SOURCE_REMOVE
+
+    def __save(self, *args):
+        GLib.idle_add(self.__idle_save)
+
+    def __check_state(self, *args):
+        if True in [
+            self.expander_cas.get_enable_expansion(),
+            self.expander_dls.get_enable_expansion(),
+            self.expander_fxaa.get_enable_expansion(),
+            self.expander_smaa.get_enable_expansion(),
+        ] or self.btn_lut_file_path is not False:
+            self.btn_save.set_sensitive(True)
+        else:
+            self.btn_save.set_sensitive(False)
+
+    def __default(self, widget, state):
+        self.expander_cas.set_sensitive(not state)
+        self.expander_dls.set_sensitive(not state)
+        self.expander_fxaa.set_sensitive(not state)
+        self.expander_smaa.set_sensitive(not state)
+        self.action_clut.set_sensitive(not state)
+        if state is False:
+            if self.expander_cas.get_enable_expansion() is False \
+                and self.expander_dls.get_enable_expansion() is False\
+                and self.expander_fxaa.get_enable_expansion() is False\
+                and self.expander_smaa.get_enable_expansion() is False\
+                and self.btn_lut_file_path is False:
+
+                self.btn_save.set_sensitive(False)
+        else:
+            self.btn_save.set_sensitive(True)
+
+    def __change_edge_detection_type(self, widget, edge_detection_type):
+        self.smaa_edge_detection = edge_detection_type
+        self.toggle_luma.handler_block_by_func(self.__change_edge_detection_type)
+        self.toggle_color.handler_block_by_func(self.__change_edge_detection_type)
+        if edge_detection_type == "luma":
+            self.toggle_color.set_active(False)
+            self.toggle_luma.set_active(True)
+        elif edge_detection_type == "color":
+            self.toggle_color.set_active(True)
+            self.toggle_luma.set_active(False)
+
+        self.toggle_luma.handler_unblock_by_func(self.__change_edge_detection_type)
+        self.toggle_color.handler_unblock_by_func(self.__change_edge_detection_type)
+
+    def __import_clut(self, *args):
+        def set_path(_dialog, response, _file_dialog):
+            if response == -3:
```

Comment:
BURN THIS WITH FIRE. Find the correct response in https://docs.gtk.org/gtk4/enum.ResponseType.html, e.g. Gtk.ResponseType.Ok,but who knows whats the correct number